portfolio/src/main/webapp/script.js

```diff
@@ -11,18 +11,98 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
- 
+
+var userLoggedIn;
+var username;
+var adminPrivileges;
+var commentStorage;
+
+// Fetches login/logout link along with login status
+function getLink(){
+  fetch('/log').then(response => response.json()).then((logInfo) => {
+    const jsonValues = Object.values(logInfo);
+
+    var logUrl = logInfo.logUrl;
+    userLoggedIn = logInfo.isUserLoggedIn;
+    username = logInfo.username;
+      
+    var loginLink = document.getElementById('loginLink');
+    loginLink.href = logUrl;
+
+    var loginStatus = document.getElementById('loginDescription');
+    if(userLoggedIn == 'true'){ 
+      var loginStatusContainer = document.getElementById('loginStatus');
+      var usernameStatus = document.createElement('p');
+      var usernameStorage = document.getElementById('nameStorage');
+
+      usernameStatus.innerHTML = 'Logged in as: ' + username;
+      loginStatusContainer.appendChild(usernameStatus);
+      usernameStorage.value = username;
+
+      if(username == 'nzbutler' || username == 'neriyahbutler21'){
+        adminPrivileges = true;
+        }else{ adminPrivileges = false;}
+
+      loginStatus.innerHTML = 'LOG OUT HERE';
+    }
+    else{ loginStatus.innerHTML = 'LOG IN HERE';} 
+  });
+}
+
+// Gathers all the comments from the json object and puts them on the website
 function getComments(){
+  getLink();
+
   fetch('/data').then(response => response.json()).then((text) => {
-    const commentContainer = document.getElementById('commentList');
     const jsonValues = Object.values(text);
- 
-    console.log(jsonValues);
-    
-    for(var i = 0; i < Object.values(text).length; i++){
-      var comment = document.createElement('p');
-      comment.innerHTML = jsonValues[i];
-      commentContainer.appendChild(comment);
+    jsonValues.forEach((task) => {
+      if(userLoggedIn == 'true'){
+        const commentsContainer = document.getElementById('commentList');
+        var comment = document.createElement('div');
+        comment.className = 'comment';
+
+        var commentInfo = document.createElement('div');
+        commentInfo.className = 'commentInfo';
+
+        var commentMain = document.createElement('div');
+        commentMain.className = 'commentMain';
+
+        var commentDetails = document.createElement('div');
+        commentDetails.className = 'commentDetails';
+
+        commentMain.innerHTML = '\"' + task.comment + '\"';
+        commentDetails.innerHTML = task.username;
+
+        commentInfo.appendChild(commentMain);
+        commentInfo.appendChild(commentDetails);
+        comment.appendChild(commentInfo);
+
+        if(adminPrivileges == true){
```

Comment:
This check might be OK, as long as it is duplicated by an equivalent check on the server side.